### PR TITLE
changed class based remoteextensions component to functional component

### LINKED ISSUE #8761: [UI] Convert `RemoteExtensions` from class-based component to functional component
### Current Behavior
`RemoteExtensions` is a class-based component

### Desired Behavior
`RemoteExtensions` should be a functional component 

### Implementation
Convert all the Class-based components to new updated functional components. It is important here to convert a class-based component to a functional component you may need to convert some functions as well. Check if the function is used in other components or not.

Also, make sure to use the hooks while converting them. For example, you need to covert withNotify with useNotification, useRouter hook. Please check if a hook for functional components is available and use that.

Here is a quick blog that will help you get started with : [Link](https://www.digitalocean.com/community/tutorials/five-ways-to-convert-react-class-components-to-functional-components-with-react-hooks)

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by EraKin575**
**Notes for Reviewers**

This PR fixes #8761 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: ui/pages/extension/[...component].js [hunk 1]
```diff
@@ -48,53 +49,99 @@ function matchComponentURI(extensionURI, currentURI) {
...
+
+  // componentDidMount() {
+  //   dataFetch(
+  //     "/api/provider/capabilities",
+  //     {
+  //       method : "GET",
+  //       credentials : "include",
+  //     },
+  //     (result) => {
+  //       this.props.updatepagepath({ path : getPath() });
```
Comment (sudhanshutech) [inline ui/pages/extension/[...component].js:95]: remove this commented out code, no need to have these

File: ui/pages/extension/[...component].js [hunk 2]
```diff
@@ -48,53 +49,99 @@ function matchComponentURI(extensionURI, currentURI) {
...
-    this.state = {
-      componentTitle : '',
-      isLoading : true,
-      capabilitiesRegistryObj : null,
+const RemoteExtension=(props) => {
+  // constructor(props) {
+  //   super(props);
+  //   this.state = {
+  //     componentTitle : '',
+  //     isLoading : true,
```
Comment (sudhanshutech) [inline ui/pages/extension/[...component].js:57]: remove

File: ui/pages/extension/[...component].js [hunk 3]
```diff
@@ -48,53 +49,99 @@ function matchComponentURI(extensionURI, currentURI) {
...
+
+  // componentWillUnmount() {
+  //   this.setState({
+  //     componentTitle : '',
+  //     isLoading : true,
+  //     capabilitiesRegistryObj : null,
+  //   })
+  // }
+  useEffect(() => {
+    prevProps.current = props;``
```
Comment (sudhanshutech) [inline ui/pages/extension/[...component].js:75]: what is the use of this `useEffect`

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

File: ui/pages/extension/[...component].js (hunk 3 above)
Comment (EraKin575) [reply ui/pages/extension/[...component].js:75]: well I put useEffect there to keep track of props that can change between renders

Comment (aabidsofi19) [reply ui/pages/extension/[...component].js]: no please ,  you dont need useEffect to track props they automatically do .

**Comment by aabidsofi19** [comment]
Please remove the redundant effects

**Comment by l5io** [comment]
🚀 Preview for commit 832945d0a62b04f6f89a76969bdbcf91ec4e619e at: https://650a0f9eb3172054ea55a5c2--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit ffa9293b90b1c761f2569d5e8761072cec137d1d at: https://650a10d6fa07be4f62c6bb7f--meshery-docs-preview.netlify.app

**Comment by saurabh100ni** [comment]
Hey @EraKin575
Let's discuss that in today's Meshery Development call.
please add this to your agenda if you would : ) to: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit

**Comment by EraKin575** [comment]
@aabidsofi19  please have a look

**Pull request closed without merging**